### PR TITLE
Add CNTV_CVAL_EL0, VBAR_EL2, OSLAR_EL1 and more fields to HCR_EL2

### PR DESCRIPTION
**Pull request by Javier-varez**
I'm adding a few registers I use in my kernel and that would hopefully be useful for other people too.

### CODE REVIEW

File: src/registers/hcr_el2.rs [hunk 1]
```diff
@@ -18,6 +19,53 @@ use tock_registers::{
...
+        ///     AUTDA, AUTDB, AUTDZA, AUTDZB, AUTIA, AUTIA1716, AUTIASP, AUTIAZ, AUTIB, AUTIB1716,
+        ///     AUTIBSP, AUTIBZ, AUTIZA, AUTIZB, PACGA, PACDA, PACDB, PACDZA, PACDZB, PACIA,
+        ///     PACIA1716, PACIASP, PACIAZ, PACIB, PACIB1716, PACIBSP, PACIBZ, PACIZA, PACIZB.
+        ///     RETAA, RETAB, BRAA, BRAB, BLRAA, BLRAB, BRAAZ, BRABZ, BLRAAZ, BLRABZ.
+        ///     ERETAA, ERETAB, LDRAA, and LDRAB.
+        API   OFFSET(41) NUMBITS(1) [
+            EnableTrapPointerAuthInstToEl2 = 0,
+            DisableTrapPointerAuthInstToEl2 = 1
+        ],
+        ///
```
Comment (andre-richter) [inline src/registers/hcr_el2.rs]:
```suggestion

```

File: src/registers/hcr_el2.rs [hunk 2]
```diff
@@ -18,6 +19,53 @@ use tock_registers::{
...
+        /// Trap registers holding "key" values for Pointer Authentication. Traps accesses to the
+        /// following registers from EL1 to EL2, when EL2 is enabled in the current Security state,
+        /// reported using EC syndrome value 0x18:
+        ///
+        ///     APIAKeyLo_EL1, APIAKeyHi_EL1, APIBKeyLo_EL1, APIBKeyHi_EL1, APDAKeyLo_EL1,
+        ///     APDAKeyHi_EL1, APDBKeyLo_EL1, APDBKeyHi_EL1, APGAKeyLo_EL1, and APGAKeyHi_EL1.
+        ///
```
Comment (andre-richter) [inline src/registers/hcr_el2.rs]:
```suggestion

```

File: src/registers/hcr_el2.rs [hunk 3]
```diff
@@ -18,6 +19,53 @@ use tock_registers::{
 
 register_bitfields! {u64,
     pub HCR_EL2 [
+        ///
```
Comment (andre-richter) [inline src/registers/hcr_el2.rs]:
```suggestion
```

Comment (andre-richter) [reply src/registers/hcr_el2.rs]: This is supposed to mean delete line. Unsure if I did that correctly.

File: src/registers/hcr_el2.rs (hunk 1 above)
Comment (andre-richter) [reply src/registers/hcr_el2.rs]: This is supposed to mean empty line.

File: src/registers/hcr_el2.rs [hunk 4]
```diff
@@ -18,6 +19,53 @@ use tock_registers::{
...
+            EnableTrapPointerAuthKeyRegsToEl2 = 0,
+            DisableTrapPointerAuthKeyRegsToEl2 = 1,
+        ],
+
+        /// Route synchronous External abort exceptions to EL2.
+        ///   if 0: This control does not cause exceptions to be routed from EL0 and EL1 to EL2.
+        ///   if 1: Route synchronous External abort exceptions from EL0 and EL1 to EL2, when EL2 is
+        ///         enabled in the current Security state, if not routed to EL3.
+        TEA   OFFSET(37) NUMBITS(1) [
+            DisableTrapSyncExtAborts = 0,
```
Comment (andre-richter) [inline src/registers/hcr_el2.rs]:
```suggestion
            DisableTrapSyncExtAbortsToEl2 = 0,
```

File: src/registers/hcr_el2.rs [hunk 5]
```diff
@@ -18,6 +19,53 @@ use tock_registers::{
...
+            EnableTrapSyncExtAborts = 1,
```
Comment (andre-richter) [inline src/registers/hcr_el2.rs]:
```suggestion
            EnableTrapSyncExtAbortsToEl2 = 1,
```

File: src/registers/hcr_el2.rs [hunk 6]
```diff
@@ -39,6 +87,40 @@ register_bitfields! {u64,
...
+        ///
+        ///       - HCR_EL2.E2H is 0, the Effective values of the HCR_EL2.{FMO, IMO, AMO} fields
+        ///         are 1.
+        ///       - HCR_EL2.E2H is 1, the Effective values of the HCR_EL2.{FMO, IMO, AMO} fields
+        ///         are 0.
+        ///
+        ///   - For further information on the behavior of this bit when E2H is 1, see 'Behavior of
+        ///     HCR_EL2.E2H'.
+        TGE   OFFSET(27) NUMBITS(1) [
+            DisableTrapGeneralExceptions = 0,
```
Comment (andre-richter) [inline src/registers/hcr_el2.rs]:
```suggestion
            DisableTrapGeneralExceptionsToEl2 = 0,
```

File: src/registers/hcr_el2.rs [hunk 7]
```diff
@@ -39,6 +87,40 @@ register_bitfields! {u64,
...
+            EnableTrapGeneralExceptions = 1,
```
Comment (andre-richter) [inline src/registers/hcr_el2.rs]:
```suggestion
            EnableTrapGeneralExceptionsToEl2 = 1,
```

File: src/registers/oslar_el1.rs [hunk 8]
```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//
+// Copyright (c) 2018-2022 by the author(s)
+//
+// Author(s):
+//   - Andre Richter <andre.o.richter@gmail.com>
+//   - Javier Alvarez <javier.alvarez@allthingsembedded.net>
+
+//! OS Lock Access Register
```
Comment (andre-richter) [inline src/registers/oslar_el1.rs]:
```suggestion
//! OS Lock Access Register - EL1
```

**Review by andre-richter — COMMENTED** [review]
Thanks a lot! 👍 

I left some minor suggestions, please have a look.

File: src/registers/hcr_el2.rs (hunk 1 above)
Comment (Javier-varez) [reply src/registers/hcr_el2.rs]: Done 👍🏻

File: src/registers/hcr_el2.rs (hunk 2 above)
Comment (Javier-varez) [reply src/registers/hcr_el2.rs]: Done 👍🏻

File: src/registers/hcr_el2.rs (hunk 3 above)
Comment (Javier-varez) [reply src/registers/hcr_el2.rs]: Done 👍🏻

File: src/registers/hcr_el2.rs (hunk 4 above)
Comment (Javier-varez) [reply src/registers/hcr_el2.rs]: Done 👍🏻

File: src/registers/hcr_el2.rs (hunk 6 above)
Comment (Javier-varez) [reply src/registers/hcr_el2.rs]: Done 👍🏻

File: src/registers/hcr_el2.rs (hunk 7 above)
Comment (Javier-varez) [reply src/registers/hcr_el2.rs]: Done 👍🏻

File: src/registers/oslar_el1.rs (hunk 8 above)
Comment (Javier-varez) [reply src/registers/oslar_el1.rs]: Done 👍🏻

File: src/registers/hcr_el2.rs (hunk 5 above)
Comment (Javier-varez) [reply src/registers/hcr_el2.rs]: Done 👍🏻

**Comment by Javier-varez** [comment]
Thanks a lot for the review! I applied the feedback as a separate commit, but let me know if you would rather have it amended into the previous changes

**Comment by andre-richter** [comment]
Thanks again!

Published and pushed to crates.io.

https://github.com/rust-embedded/cortex-a/releases/tag/v7.2.0

**Pull request merged**